backend/src/ml_space_lambda/dataset/lambda_functions.py

```diff
@@ -124,6 +124,10 @@ def get(event, context):
     dataset_name = event["pathParameters"]["datasetName"]
 
     dataset = dataset_dao.get(scope, dataset_name)
+    if scope == DatasetType.GROUP:
+        groups = group_dataset_dao.get_groups_for_dataset(dataset_name)
+        for group in groups:
+            dataset.groups.append(group.group)
```

Comment:
Something to be cognizant of is that we're returning this dataset with the `groups` field filled in. Although it doesn't currently happen (to my knowledge) the `edit` method above takes the dataset object and persists it as-is. If this ever came back with these groups the next time it is fetched we'd be appending duplicates groups. At a minimum I think we should be doing something like `dataset.groups = list(set([group.name for group in groups]))`

Reply:
So this actually isn't a problem because of how we update the DDB entry in the `edit` function:

```
exp_values = json.loads(
            dynamodb_json.dumps(
                {
                    ":description": dataset.description,
                    ":lastUpdatedAt": time.time(),
                    ":name": name,
                    ":scope": scope,
                }
            )
        )
```

We never load the `groups` field into the update so it never makes it to the table.